tests/tracing/test_skeletonize.py

```diff
@@ -0,0 +1,281 @@
+
+CIRCULAR_TARGET = np.array(
+    [
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+    ]
+)
+
+
+def test_skeletonize_method(skeletonize_circular_bool_int: np.ndarray) -> None:
+    """Test unsupported method raises the appropriate error."""
+    with pytest.raises(ValueError):
+        get_skeleton(skeletonize_circular_bool_int, method="nonsense")
+
+
+def test_skeletonize_circular_zha(skeletonize_circular_bool_int: np.ndarray) -> None:
+    """Test the Zhang method of skeletionzation on a circular object."""
+    test = get_skeleton(skeletonize_circular_bool_int, method="zhang").astype(int)
+    assert isinstance(test, np.ndarray)
+    assert test.ndim == 2
+    assert test.shape == (21, 21)
+    assert test.sum() == 36
+    np.testing.assert_array_equal(test, CIRCULAR_TARGET)
+
+
+def test_skeletonize_circular_lee(skeletonize_circular_bool_int: np.ndarray) -> None:
+    """Test the Lee method of skeletonization on a circular object."""
+    test = get_skeleton(skeletonize_circular_bool_int, method="lee").astype(int)
+    assert isinstance(test, np.ndarray)
+    assert test.ndim == 2
+    assert test.shape == (21, 21)
+    assert test.sum() == 36
+    np.testing.assert_array_equal(test, CIRCULAR_TARGET)
+
+
+def test_skeletonize_circular_medial_axis(skeletonize_circular_bool_int: np.ndarray) -> None:
+    """Test the medial axis method of skeletonization on a circular object."""
+    CIRCULAR_MEDIAL_AXIS_TARGET = np.array(
```

Comment:
This array looks unusual (not what I'd expect from skeletonising a square), have we tested this method on anything else to see if its results are useful for our data? If not we might not need it.

Reply:
Yes, I wasn't sure if this method would be useful but it is what the method of [medial axis](https://homepages.inf.ed.ac.uk/rbf/HIPR2/skeleton.htm) should look like (if I've understood the method correctly).

Happy to have it removed, but didn't see any harm in including it, perhaps it will be worth noting with a WARNING when I get round to writing documentation.